light/light_test.go

```diff
@@ -136,3 +142,18 @@ func TestClientIntegration_VerifyLightBlockAtHeight(t *testing.T) {
 
 	require.EqualValues(t, 3, h.Height)
 }
+
+func waitForBlock(ctx context.Context, p provider.Provider, height int64) (*types.LightBlock, error) {
+	for {
+		block, err := p.LightBlock(ctx, height)
+		switch err {
+		case nil:
+			return block, nil
+		// node isn't running yet, wait 1 second and repeat
+		case provider.ErrNoResponse, provider.ErrHeightTooHigh:
+			time.Sleep(1 * time.Second)
```

Comment:
I'd do this as a ticker/timer and have this happen inside of a select statement with the timer/ticker and the ctx.Done() in different cases.